plugin-server/src/ingestion/ingestion-consumer.test.ts

```diff
@@ -691,4 +706,459 @@ describe('IngestionConsumer', () => {
             TRANSFORMATION_TEST_TIMEOUT
         )
     })
+
+    describe('transformation chains', () => {
+        beforeEach(async () => {
+            fixedTime = DateTime.fromObject({ year: 2025, month: 1, day: 1 }, { zone: 'UTC' })
+            jest.spyOn(Date, 'now').mockReturnValue(fixedTime.toMillis())
+
+            hub = await createHub()
+            await resetTestDatabase()
+
+            // Create team first before inserting hog functions
+            const team = await getFirstTeam(hub)
+
+            // Set up GeoIP database
+            const mmdbBrotliContents = readFileSync(join(__dirname, '../../tests/assets/GeoLite2-City-Test.mmdb.br'))
+            hub.mmdb = Reader.openBuffer(brotliDecompressSync(mmdbBrotliContents))
+
+            hub.kafkaProducer = mockProducer
+
+            // Create ingester
+            ingester = new IngestionConsumer(hub)
+
+            // Start hogFunctionManager with correct types before starting ingester
+            await ingester.start()
+
+            // Set up the transformations in order
+            const transformations: Partial<HogFunctionType>[] = [
```

Comment:
This whole thing could be more like:

In general we should write as clean and tidy tests as possible as otherwise we will get annoyed by them and not update them as we should

```
[
  [botDetectionTemplate, INPUTS],
  [posthogFilterOutPlugin, INPUTS]
].map(([template, inputs]) => ({
  id: new UUIDT().toString(),
  name: template.name,
  ...
})
```